buildSrc/src/main/java/org/elasticsearch/gradle/DockerUtils.java

```diff
@@ -0,0 +1,228 @@
+
+        if (imagesResult.isSuccess() == false) {
+            dockerAvailability.lastCommand = versionResult;
+            return dockerAvailability;
+        }
+
+        dockerAvailability.isAvailable = true;
+        return dockerAvailability;
+    }
+
+    /**
+     * This class represents the results of a Docker search from {@link #getDockerAvailability()}}.
+     */
+    public static class DockerAvailability {
+        private boolean isAvailable = false;
+        private boolean isVersionHighEnough = false;
+        private String path = null;
+        private String version = null;
+        private Result lastCommand = null;
+
+        /**
+         * Indicates whether Docker is available and meets the required criteria.
+         * @return true if, and only if, Docker is:
+         * <ul>
+         *     <li>Installed</li>
+         *     <li>Executable</li>
+         *     <li>Is at least version 17.05</li>
+         *     <li>Can execute a command that requires privileges</li>
+         * </ul>
+         */
+        public boolean isAvailable() {
+            return isAvailable;
+        }
+
+        /**
+         * @return true if the installed Docker version is &gt;= 17.05
+         */
+        public boolean isVersionHighEnough() {
+            return isVersionHighEnough;
+        }
+
+        /**
+         * @return the path to the Docker CLI, or null
+         */
+        public String getPath() {
+            return path;
+        }
+
+        /**
+         * @return The installed Docker version, or null
+         */
+        public String getVersion() {
+            return version;
+        }
+
+        /**
+         * @return Information about the last command executes while probing Docker, or null.
+         */
+        public Result getLastCommand() {
+            return lastCommand;
+        }
+    }
+
+    // package-private for testing
+    /**
+     * Given the string that <code>docker --version</code> prints, extract the docker version from it.
+     * @param dockerVersion the version string to parse
+     * @return the Docker version
+     */
+    static String extractVersion(String dockerVersion) {
+        final Pattern dockerVersionPattern = Pattern.compile("Docker version (\\d+\\.\\d+\\.\\d+(?:-[a-zA-Z0-9]+)?), build [0-9a-f]{7,40}");
+
+        final Matcher matcher = dockerVersionPattern.matcher(dockerVersion);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("Can't extract version from: [" + dockerVersion + "]");
+        }
+
+        return matcher.group(1);
+    }
+
+    // package-private for testing
+    /**
+     * Checks whether the supplied Docker version is &gt;= 17.05
+     * @param dockerVersion the version to check
+     * @return true if the version is high enough
+     */
+    static boolean checkVersion(String dockerVersion) {
+        final String[] majorMinor = dockerVersion.split("\\.");
```

Comment:
I think we could use `org.elasticsearch.gradle.Version` here, which would allow us to model the minimum required version as a `static Version` constant in this class rather than hard-code it in this method implementation.

Reply:
Ah, I didn't know about that class, that's a lot better.